checks/sast.go

```diff
@@ -61,52 +61,50 @@ func SAST(c *checker.CheckRequest) checker.CheckResult {
 	if sastScore != checker.InconclusiveResultScore &&
 		codeQlScore != checker.InconclusiveResultScore {
 		switch {
-		// This only happens if:
+		// sastScore >= codeQlScore only happens if:
 		// - sastScore is maximum and codeQl is enabled OR
 		// - sastScore is minimum and codeQl is not enabled
-		// In both cases, sastReason gives the best reason to the user.
-		case sastScore >= codeQlScore:
-			// Add codeQlReason to the details.
-			c.Dlogger.Warn(codeQlReason)
-			return checker.CreateProportionalScoreResult(CheckSAST, sastReason, sastScore, checker.MaxResultScore)
+		case sastScore == checker.MaxResultScore && sastScore >= codeQlScore:
+			return checker.CreateProportionalScoreResult(CheckSAST,
+				"all commmits are checked with a SAST tool", sastScore, checker.MaxResultScore)
+		case sastScore != checker.MaxResultScore && sastScore >= codeQlScore:
+			return checker.CreateMinScoreResult(CheckSAST,
+				"no SAST tool detected")
 		// codeQl is enabled and sast has 0+ (but not all) PRs checks.
-		// In this case, codeQlReason provides good information.
 		case codeQlScore == checker.MaxResultScore:
-			// Add sastReason to the details.
-			c.Dlogger.Warn(sastReason)
 			const sastWeight = 3
 			const codeQlWeight = 7
 			score := checker.AggregateScoresWithWeight(map[int]int{sastScore: sastWeight, codeQlScore: codeQlWeight})
-			return checker.CreateResultWithScore(CheckSAST, codeQlReason, score)
+			return checker.CreateResultWithScore(CheckSAST, "SAST tool detected but not used on all commmits", score)
 		default:
 			return checker.CreateRuntimeErrorResult(CheckSAST, sce.Create(sce.ErrScorecardInternal, "contact team"))
 		}
 	}
 
 	// CodeQl inconclusive.
+	// Can never happen currently.
 	if codeQlScore != checker.InconclusiveResultScore {
-		c.Dlogger.Warn(sastReason)
-		return checker.CreateResultWithScore(CheckSAST, codeQlReason, codeQlScore)
+		return checker.CreateResultWithScore(CheckSAST, "internal error running codeQl test", codeQlScore)
 	}
 
 	// Sast inconclusive.
+	// Only happens if no merges are detected.
 	if sastScore != checker.InconclusiveResultScore {
-		c.Dlogger.Warn(codeQlReason)
-		return checker.CreateResultWithScore(CheckSAST, sastReason, sastScore)
+		return checker.CreateResultWithScore(CheckSAST, "no merges detected", sastScore)
```

Comment:
what kind of merges, should be better to explain it more, otherwise feels incomplete error message.

Reply:
see https://github.com/ossf/scorecard/pull/760